plugins/backstage-plugin-coder/docs/guides/sdk-recommendations.md

```diff
@@ -0,0 +1,557 @@
+  queryKey: ['workspaces'],
+
+  // Access the SDK without needing to import or call useCoderSdk
+  queryFn: ({ sdk }) => sdk.getWorkspaces({ q: 'owner:me' }),
+});
+```
+
+## Authentication
+
+All API methods from the SDK will throw an error if the user is not authenticated. The Coder plugin provides a few different ways of letting the user authenticate with Coder:
+
+- The official Coder components
+- The `CoderProvider` component's fallback auth UI
+- The `useCoderAuth` hook
+
+All three solutions, directly or indirectly, involve the `CoderAuth` type. More information can be found under the `useCoderAuth` section.
+
+### Authenticating via official Coder components
+
+Every official Coder component (such as `CoderWorkspacesCard`) exported through the plugin is guaranteed to have some mechanism for supplying auth information. This is typically done via a UI form.
+
+<-- [Include screenshot/video of auth input here] -->
+
+#### Pros
+
+- Come pre-wired with the ability to let the user supply auth information
+- Extensively tested
+- Let you override the styles or re-compose the individual pieces for your needs
+- Always audited for WCAG Level AA accessibility, and include landmark behavior for screen readers
+
+#### Cons
+
+- Not every Coder component makes sense for every page
+- No easy mechanism for ensuring that your custom components don't run until the user authenticates via the official Coder components
+- Components only work with a small sub-section of the total API, and won't be able to satisfy true power users
+- Must be mounted within a `CoderProvider` component
+
+### Authenticating via the fallback auth UI
+
+When you include the `CoderProvider` component in your Backstage deployment, you have the option to set the value of `fallbackAuthUiMode`. This value affects how `CoderProvider` will inject a fallback auth input into the Backstage deployment's HTML. This means that, even if you don't use any Coder components, or are on a page that can't use them, users will always have some way of supplying auth information.
+
+```tsx
+<CoderProvider appConfig={youCustomAppConfig} fallbackAuthUiMode="assertive">
+  <RestOfYourComponentsThatUseTheCoderApis />
+</CoderProvider>
+```
+
+<-- [Include screenshot/video of fallback auth input here] -->
+
+The fallback auth UI will never be visible while the user is authenticated. However, if the user is **not** authenticated, then the value of `fallbackAuthUiMode` will affect what appears on screen:
+
+- `restrained` (default) - The fallback auth UI will not appear if there are official Coder components on screen.
+- `hidden` - The fallback auth is **never** visible on the page. If no official Coder components are on screen, you will need to import `useCoderAuth` into your custom components to authenticate your users.
+- `assertive` - The fallback auth UI is always visible when the user is not authenticated, regardless of whether there are any official Coder components on screen.
+
+#### Pros
```

Comment:
I am struggling to find the information I care most about in this README so that I can get up and moving quickly. Do you think we could find some ways to shorten it for clarity? One idea is that was could fold important callouts from the `Pros` and `Cons` headers in each section into the section body itself.

Reply:
Alternatively, we could make a table and link for all 3 with pros and cons to an example component for each (maybe even just linking to the test case is enough)? Agree that someone should be able to weigh pros and cons at a glance and also see an example for each so that can see what fits into what they are trying to achieve.